### ISSUE TITLE
Citation Styles have new line after starting numeric in Entry Preview. Differs from OO Cite.

### ISSUE THREAD

**Issue by ThiloteE** [subhramit/jabref#10]
### JabRef version

Latest development branch build (please note build date below)

### Operating system

Windows

### Details on version and operating system

Windows 10

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

First-logic branch commit ba872be2789ddcee7a693cfc7fa705e1a4b8e454

Select a style and look at entry preview.
![image](https://github.com/user-attachments/assets/7e11c6cc-64c8-4b4f-b0c5-39e38d4392cf)

It is assumed there is not supposed to be a new line. Here is the corresponding citation in LibreOffice:

![image](https://github.com/user-attachments/assets/75e2f439-0735-488d-b3a8-22f562a4f1b5)

**Comment by subhramit** [comment]
Hi Thilo, could you shift this issue to the main repo?
I initially thought it's of the level of a "good first issue", tried investigating, but couldn't find what's causing it.
This could mean either I'm missing something obvious, or the bug is deeper down.

**Comment by ThiloteE** [comment]
Waiting for Olivers opinion. If I remember correctly, you both did talk about this topic before, right?

**Comment by subhramit** [comment]
> Waiting for Olivers opinion. If I remember correctly, you both did talk about this topic before, right?

No, we didn't. I had noticed this, but the first time we talked about this was during our last exchange call (Oliver wasn't there).

Reason I suggested to shift is because the issue exists independent of my changes (it was there in the entry editor of the preferences even before our project began).

**Comment by koppor** [comment]
Yes, we talked about it. Maybe one or two months ago.

@subhramit needs to find out how to deal. Write JUnit tests. Try to craft a minimal example. org.jabref.logic.citationstyle.CitationStyleGeneratorTest is the class to work on. Then adapt org.jabref.logic.citationstyle.CitationStyleGenerator to not have the line break.

**Comment by koppor** [comment]
> Reason I suggested to shift is because the issue exists independent of my changes (it was there in the entry editor of the preferences even before our project began).

You are learning that one is dependent on other code, which also needs to be adapted. Fortunately, as we are open source, you can do it for yourself and do not need to wait for others to take action. 💪

**Comment by subhramit** [comment]
> > Reason I suggested to shift is because the issue exists independent of my changes (it was there in the entry editor of the preferences even before our project began).
> 
> You are learning that one is dependent on other code, which also needs to be adapted. Fortunately, as we are open source, you can do it for yourself and do not need to wait for others to take action. 💪

Hey, yes, I'd be more than happy to do that, by the time I had written the first comment, I had already tried debugging it:

> Hi Thilo, could you shift this issue to the main repo? I initially thought it's of the level of a "good first issue", tried investigating, but couldn't find what's causing it. This could mean either I'm missing something obvious, or the bug is deeper down.



> @subhramit needs to find out how to deal. Write JUnit tests. Try to craft a minimal example. org.jabref.logic.citationstyle.CitationStyleGeneratorTest is the class to work on. Then adapt org.jabref.logic.citationstyle.CitationStyleGenerator to not have the line break.

The issue, I think, is not with `CitationStyleGenerator`, but `PreviewViewer` (I am also using `CitationStyleGenerator` to generate CSL Styles for LO, and none of the outputs produced have the line break, whether after processing, or in the raw HTML). `PreviewViewer` is also using HTML-format generated citations from `CitationStyleGenerator`, same as I am using.

Furthermore, speaking of `CitationStyleGeneratorTest`, if you look at the expected values, none of the numbered citations have a line break after the number, so again it cannot be `CitationStyleGenerator`.

In `PreviewViewer`, this is where I think the issue is happening:
![image](https://github.com/user-attachments/assets/853cc610-f07d-4bbb-8076-582f1902de10)
But don't know how to go deeper than this for the fix... (anything I ctrl+click on deeper than this I get decompiled `.class`es).

**Comment by ThiloteE** [comment]
Likely depends on Javafx. https://openjfx.io/javadoc/21/javafx.web/javafx/scene/web/WebView.html

**Comment by subhramit** [comment]
TODO: File issue upstream at JavaFX (WebScene).

**Comment by koppor** [comment]
> TODO: File issue upstream at JavaFX (WebScene).

I did not see any MWE for this. Requested at https://github.com/subhramit/jabref/issues/10#issuecomment-2241776572. Such an MWE ensures that the "right" component is identified before filing any issue. Moreover, after a fix, one can directly see if any other component broke. Finally, one can work on work arounds... Working on a minimal example is easier, because there is no unnecessary code executed (which takes time and distracts during debugging).
